sdk/service.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package prover;
+
+service SP1ProverService {
+  rpc CreateProof(CreateProofRequest) returns (CreateProofResponse) {}
+  rpc SubmitProof(SubmitProofRequest) returns (SubmitProofResponse) {}
+  rpc GetProofStatus(GetProofStatusRequest) returns (GetProofStatusResponse) {}
+}
+
```

Comment:
Documenting each of these states (what they mean, how they transition) would be good